packages/devextreme/js/__internal/ui/shared/m_grouped_data_converter_mixin.ts

```diff
@@ -1,49 +1,60 @@
 import { isObject } from '@js/core/utils/type';
+import type { DataSourceLike, DataSourceOptions } from '@js/data/data_source';
+import type { GroupedItem } from '@ts/ui/list/list.edit.strategy.grouped';
+
+type ConvertedDataSourceLike<TItem> = DataSourceLike<TItem>
+  & { group?: DataSourceOptions['group'] & { keepInitialKeyOrder?: boolean } };
+
+const hasCorrectStructure = (
```

Comment:
usual naming for type guard - isTypeName
```suggestion
const isGroupedItem = (
```